include/deal.II/base/polynomials_bernardi_raugel.h

```diff
@@ -0,0 +1,210 @@
+   */
+  unsigned int
+  n() const;
+
+
+  /**
+   * Return the degree of Bernardi-Raugel polynomials.
+   * Since the bubble functions are quadratic in at least one variable,
+   * the degree of the Bernardi-Raugel polynomials is two.
+   */
+  unsigned int
+  degree() const;
+
+  /**
+   * Return the name of the space, which is <tt>BernardiRaugel</tt>.
+   */
+  std::string
+  name() const;
+
+  /**
+   * Compute the value and derivatives of each Bernardi-
+   * Raugel polynomial at @p unit_point.
+   *
+   * The size of the vectors must either be zero or equal <tt>n()</tt>.  In
+   * the first case, the function will not compute these values.
+   *
+   * If you need values or derivatives of all tensor product polynomials then
+   * use this function, rather than using any of the <tt>compute_value</tt>,
+   * <tt>compute_grad</tt> or <tt>compute_grad_grad</tt> functions, see below,
+   * in a loop over all tensor product polynomials.
+   */
+  void
+  compute(const Point<dim> &           unit_point,
+          std::vector<Tensor<1, dim>> &values,
+          std::vector<Tensor<2, dim>> &grads,
+          std::vector<Tensor<3, dim>> &grad_grads,
+          std::vector<Tensor<4, dim>> &third_derivatives,
+          std::vector<Tensor<5, dim>> &fourth_derivatives) const;
+  /**
+   * Return the number of polynomials in the space <tt>BR(degree)</tt> without
+   * requiring to build an object of PolynomialsBernardiRaugel. This is
+   * required by the FiniteElement classes.
+   */
+  static unsigned int
+  compute_n_pols(const unsigned int k);
+
+
+private:
+  /**
+   * The degree of this object given to the constructor (must be 1).
+   */
+  const unsigned int my_degree;
+
+  /**
+   * The number of Bernardi-Raugel polynomials.
+   */
+  const unsigned int n_pols;
+
+  /**
+   * An object representing the polynomial space of Q
+   * functions which forms the <tt>BR</tt> polynomials through
+   * outer products of these with the corresponding unit ijk
+   * vectors.
+   */
+  const AnisotropicPolynomials<dim> polynomial_space_Q;
+  /**
+   * An object representing the polynomial space of bubble
+   * functions which forms the <tt>BR</tt> polynomials through
+   * outer products of these with the corresponding normals.
+   */
+  const AnisotropicPolynomials<dim> polynomial_space_bubble;
+
+  /**
+   * A static member function that creates the polynomial space we use to
+   * initialize the #polynomial_space_Q member variable.
+   */
+  static std::vector<std::vector<Polynomials::Polynomial<double>>>
+  create_polynomials_Q();
+
+  /**
+   * A static member function that creates the polynomial space we use to
+   * initialize the #polynomial_space_bubble member variable.
+   */
+  static std::vector<std::vector<Polynomials::Polynomial<double>>>
+  create_polynomials_bubble();
+};
+
+
+template <int dim>
+inline unsigned int
+PolynomialsBernardiRaugel<dim>::n() const
+{
+  return n_pols;
+}
+
+template <int dim>
```

Comment:
Please use three blank lines between function definitions.

Reply:
Okay I've added them in!